### PR TITLE
Upgrade nitf spec

### PR DESCRIPTION
**Pull request by benoitrazet**
This PR proposes to upgrade the main nitf spec with the recent changes done on `nitf/nitf-modern`.

The most important changes are:
* eliminate multiple parse solutions
* spec uses only biased choice
* spec uses more recent DaeDaLus coding style
* support C++ code generation
* regression testing (`./regressions.sh`) with expected output stored in `nitf/regression_nitf.txt`

### CODE REVIEW

File: nitf/nitf_graphic_subheader.ddl [hunk 1]
```diff
@@ -9,52 +9,51 @@ def GraphicColor = Choose {
...
-
-  sxshd = Choose {
-    tre = {
-      seq = sxshdl is taggedrec ;
-      Many (seq as! uint 64) Byte
-    } ;
-    ommitted = sxshdl is notre
-  } ;
-}
+  sxshdl = Choose { -- NOTE-MODERN: nonoverlapping
```
Comment (wrharris) [inline nitf/nitf_graphic_subheader.ddl:42]: Can this and other uses of `Choose` result in ambiguity, or is there something about their context that intuitively ensures global unambiguity?

File: nitf/nitf_header.ddl [hunk 2]
```diff
@@ -47,23 +47,25 @@ def DigraphSeq = Choose1 {
...
 -- Unclear how to pad single characters here...
-def FSDCTP = Choose {
-  dd = Match "DD" ;
-  de = Match "DE" ;
-  gd = Match "GD" ;
-  ge = Match "GE" ;
-  o = Match "O " ;
-  x = Match "X " ;
-  none = Match "  " ;
+def FSDCTP = Choose { -- NOTE-MODERN: nonoverlapping
```
Comment (wrharris) [inline nitf/nitf_header.ddl:61]: Why are these non-overlapping?

File: nitf/nitf_graphic_subheader.ddl [hunk 3]
```diff
@@ -9,52 +9,51 @@ def GraphicColor = Choose {
...
+      notre = @(IsNum 5 0) ;
+      taggedrec = BoundedNum 5 3 9741
+    }
+
+  xssofl = Choose { -- NOTE-MODERN: This may overlap!!!
```
Comment (wrharris) [inline nitf/nitf_graphic_subheader.ddl:47]: Can this and other uses of `Choose` result in ambiguity, or is there something about their context that intuitively ensures global unambiguity?

File: nitf/nitf_text_subheader.ddl [hunk 4]
```diff
@@ -15,29 +15,26 @@ def TxtFmt = Choose {
...
-  txshdl = TxShDL as! uint 64;
+  txshdl = TxShDL as! uint 64
 
-  Choose {
-    txsofl = {
-      UnsignedNum 3 ;
-    } ;
-    omitted = Guard (txshdl == 0);
-  } ;
+  Choose { -- NOTE-MODERN: seems overlapping but apparently not harmful
```
Comment (wrharris) [inline nitf/nitf_text_subheader.ddl:35]: "harmful" = "causing global ambiguity"?

File: nitf/nitf_img_subheader.ddl [hunk 5]
```diff
@@ -48,148 +47,140 @@ def NRows = PosNumber 8
 
 def NCols = PosNumber 8
 
-def PVType = Choose {
-  integer = @(PadWSpaces 3 (Match "INT")) ;
-  bilevel = @(PadWSpaces 3 (Match "B")) ;
-  signed = @(PadWSpaces 3 (Match "SI")) ;
-  real = @(PadWSpaces 3 (Match "R")) ;
-  complex = @(PadWSpaces 3 (Match "C")) ;
+def PVType = Choose { -- NOTE-MODERN: nonoverlapping
```
Comment (wrharris) [inline nitf/nitf_img_subheader.ddl:50]: Which clauses overlap here?

File: nitf/nitf_lib.ddl [hunk 6]
```diff
@@ -141,126 +155,126 @@ def Leq x y = (Eq x y) | (Lt x y)   -- XXX: Why not use <= ?
 
 def PartialEq (x : OrBytes) (y : OrBytes) =
```
Comment (wrharris) [inline nitf/nitf_lib.ddl:156]: In principle, it seems better to make this a pure computation, but it's not essetial.

File: nitf/nitf_graphic_subheader.ddl (hunk 1 above)
Comment (benoitrazet) [reply nitf/nitf_graphic_subheader.ddl:42]: Now we've tracked down the places that created ambiguity on the testsuite without being too agressive, it makes sense to do the next step and use bias choice. I'll do another pass.

**Pull request merged**